nengo/utils/lock.py

```diff
@@ -0,0 +1,51 @@
+import errno
+import os
+import os.path
+import time
+
+
+class TimeoutError(Exception):
```

Comment:
Note to whoever rebases this to master: this exception should be moved to `nengo/exceptions.py`.